### ISSUE TITLE
Notebook testing preparation

### ISSUE THREAD

**Issue by BillyZhaohengLi** [illinoisdata/kishu#49]
20 data analysis/ML notebooks from Kaggle for testing.
Notebook preparation instructions:
1. Add the notebook to the [notebook](https://github.com/illinoisdata/elastic-notebook/tree/main/elastic/experiment/notebooks) directory.
2. Delete all cell/line magics.
3. Modify (either using the experiment script or manually) the data path to read from `/data/elastic-notebook/data`.
4. Download the required packages into the virtualenv at `/data/elastic-notebook/venv`.

List of current notebooks (tick once the notebook is prepared for testing):
- [ ] **GPU** [XGBoost Starter - [0.793]](https://www.kaggle.com/code/cdeotte/xgboost-starter-0-793)
- [ ] **GPU** [NBME / Deberta-base baseline [train]](https://www.kaggle.com/code/yasufuminakama/nbme-deberta-base-baseline-train)
- [ ] **GPU** [Scaling Up: Road to the Top, Part 3](https://www.kaggle.com/code/jhoward/scaling-up-road-to-the-top-part-3)
- [ ] **GPU** [GCViT: Global Context Vision Transformer](https://www.kaggle.com/code/awsaf49/gcvit-global-context-vision-transformer)
- [x] [AI4Code - Language Detection and Model Tuning](https://www.kaggle.com/code/andradaolteanu/ai4code-language-detection-and-model-tuning)
- [ ] **Incorrect notebook** [Visual In-Depth EDA – VinBigData Competition Data](https://www.kaggle.com/code/dschettler8845/visual-in-depth-eda-vinbigdata-competition-data)
- [ ] **GPU** [Sartorius - Starter Torch Mask R-CNN [LB=0.273]](https://www.kaggle.com/code/julian3833/sartorius-starter-torch-mask-r-cnn-lb-0-273)
- [ ] **GPU** [SWA, Apex AMP & Interpreting Transformers in Torch](https://www.kaggle.com/code/rhtsingh/swa-apex-amp-interpreting-transformers-in-torch)
- [ ] **GPU** [YoloX training pipeline COTS dataset [LB 0.507] !!](https://www.kaggle.com/code/remekkinas/yolox-training-pipeline-cots-dataset-lb-0-507)
- [ ] **TPU** [DevFest2020IdentifyingMelanomaInLesionImages](https://www.kaggle.com/code/usharengaraju/devfest2020identifyingmelanomainlesionimages/comments)
- [ ] **GPU** [NLP - EDA, Bag of Words, TF IDF, GloVe, BERT](https://www.kaggle.com/code/vbmokin/nlp-eda-bag-of-words-tf-idf-glove-bert)
- [ ] **TPU** [Deep Learning For NLP: Zero To Transformers & BERT](https://www.kaggle.com/code/tanulsingh077/deep-learning-for-nlp-zero-to-transformers-bert)
- [ ] **GPU** [Segmentation in PyTorch using convenient tools](https://www.kaggle.com/code/artgor/segmentation-in-pytorch-using-convenient-tools)
- [ ] **TPU** [Plant Pathology 2020 : EDA + Models](https://www.kaggle.com/code/tarunpaparaju/plant-pathology-2020-eda-models)
- [ ] **GPU** [A look at different embeddings.!](https://www.kaggle.com/code/sudalairajkumar/a-look-at-different-embeddings/notebook)

List of GPU-less notebooks:
Complete  Good example
- [x] [x] https://www.kaggle.com/code/samuelcortinhas/spaceship-titanic-a-complete-guide/notebook
- [x] [x] https://www.kaggle.com/code/dschettler8845/uwm-gi-tract-image-segmentation-eda/notebook (Notebook mentions work in progress)
- [x] [  ] https://www.kaggle.com/code/shtrausslearning/building-an-asset-trading-strategy/notebook
- [x] [x] https://www.kaggle.com/code/andreshg/nlp-glove-bert-tf-idf-lstm-explained/notebook
- [x] [x] https://www.kaggle.com/code/shahules/basic-eda-cleaning-and-glove/notebook
- [x] [x] https://www.kaggle.com/code/dimitreoliveira/model-stacking-feature-engineering-and-eda/notebook
- [x] [  ] https://www.kaggle.com/code/corochann/ashrae-training-lgbm-by-meter-type/notebook
- [x] [  ] https://www.kaggle.com/code/ekrembayar/store-sales-ts-forecasting-a-comprehensive-guide/notebook
- [x] [x] [NFL Data bowl EDA](https://www.kaggle.com/code/kalilurrahman/nfl-data-bowl-2023-offensive-plays-eda/notebook)
- [x] [x] [Time Series forecasting with Prophet](https://www.kaggle.com/code/robikscube/time-series-forecasting-with-prophet)
- [x] [  ] [AI4Code Detailed EDA](https://www.kaggle.com/code/odins0n/ai4code-detailed-eda)
- [x] [x] [Approaching (Almost) Any NLP Problem on Kaggle](https://www.kaggle.com/code/abhishek/approaching-almost-any-nlp-problem-on-kaggle)
- [x] [  ] [MSCI EDA which makes sense](https://www.kaggle.com/code/ambrosm/msci-eda-which-makes-sense)
- [x] [  ] https://www.kaggle.com/code/gpreda/lanl-earthquake-eda-and-prediction
- [x] [  ] https://www.kaggle.com/code/ohseokkim/house-price-all-about-house-price
- [x] [x] https://www.kaggle.com/code/sudalairajkumar/simple-exploration-notebook-zillow-prize/notebook
- [x] [  ] https://www.kaggle.com/code/hasanbasriakcay/tpsmay22-insightful-eda-fe-baseline/data
- [x] [  ] https://www.kaggle.com/code/isaienkov/riiid-answer-correctness-prediction-eda-modeling/notebook
- [x] [  ] https://www.kaggle.com/code/ammarnassanalhajali/riiid-lgbm-bagging2-sakt-0-781/notebook
- [x] [  ] https://www.kaggle.com/code/lordozvlad/tps-mar-fast-workflow-using-scikit-learn-intelex/data

Deprecated:
- [x] [Bruteforce Clustering](https://www.kaggle.com/code/thedevastator/bruteforce-clustering)
- [ ] https://www.kaggle.com/code/allunia/pulmonary-dicom-preprocessing/notebook

Jupyter lab notes:
A Jupyter lab instance can be started as follows to test notebooks:
1. Activate the virtualenv at `/data/elastic-notebook/venv`.
2. Add the project directory to PYTHONPATH: `export PYTHONPATH="${PYTHONPATH}:/path/to/elastic-notebook"`
3. Start jupyterlab: `jupyter lab --port=<PORT>`. An access code will be printed into the terminal; this is needed for step 5.
4. `ssh -L 8080:localhost:<PORT> netid@yongjoo-serv-01.cs.illinois.edu`
5. Open localhost:8080 in your web browser. Paste the access code from step 3 when prompted.
